arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -710,6 +710,12 @@ public fun <A> Iterable<A>.combineAll(MA: Monoid<A>): A = MA.run {
   }
 }
 
+public fun <A> Iterable<A>.combineAllOrNull(SG: Semigroup<A>): A? = SG.run {
```

Comment:
seeing this makes me think if the issue here is that what we call `combine` the std lib calls `reduce` in collections. If our semigroup op was called `reduce`, this and other functions that end up implemented in terms of reduce would be redundant. 

Given `Semigroup<A>` is a fun interface observing shape `(A, A) -> A` isn't `combineAllOrNull` the same as `reduceOrNull` ?

Reply:
> Given Semigroup<A> is a fun interface observing shape (A, A) -> A isn't combineAllOrNull the same as reduceOrNull ?

Yes, I think you're correct!